movement/roi/conditions.py

```diff
@@ -0,0 +1,105 @@
+"""Functions for computing condition arrays involving RoIs."""
+
+from collections import defaultdict
+from collections.abc import Sequence
+
+import numpy as np
+import xarray as xr
+
+from movement.roi.base import BaseRegionOfInterest
+
+
+def compute_region_occupancy(
+    data,
+    regions: Sequence[BaseRegionOfInterest],
+) -> xr.DataArray:
+    """Return a condition array indicating if points were inside regions.
+
+    The function returns a boolean DataArray where each element indicates
+    whether a point in the input ``data`` lies within the corresponding RoIs
+    in ``regions``. The original dimensions of ``data`` are preserved, except
+    for the ``space`` dimension which is replaced by the ``occupancy``
+    dimension. The ``occupancy`` dimension has a number of elements equal to
+    the number of RoIs in the ``regions`` argument and it's coordinate names
+    correspond to the names of the given RoIs.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        Spatial data to check for inclusion within the ``regions``. Must be
+        compatible with the ``data`` argument to :func:`contains_point\
+        <movement.roi.base.BaseRegionOfInterest.contains_point`.
+    regions : Sequence[BaseRegionOfInterest]
+        Regions of Interest that the points in ``data`` will be checked
+        against, to see if they lie inside.
+
+    Returns
+    -------
+    xarray.DataArray
+        A boolean DataArray that matches the dimensions of ``data``, except
+        for the ``space`` dimension which is dropped, and the addition of the
+        ``occupancy`` dimension that has the same length as the number
+        of regions provided. Coordinates along the ``occupancy`` dimension
+        match the names of the ``regions``. Values along this dimension are
+        boolean, indicating whether each point in ``data`` lies within the
+        corresponding region.
```

Comment:
This description of the returned array repeats, almost verbatim, information that's already higher up in the docstring. I would keep the description above the parameters, and just have a short one-line description for the returned array.

Reply:
```suggestion
        A boolean ``DataArray`` providing occupancy information.
```